From 8b88f660851328dc8b383aeaf101a48dfc247dba Mon Sep 17 00:00:00 2001
From: Jarkko Koivikko <jarkko.koivikko@code-q.fi>
Date: Wed, 25 Aug 2021 16:42:32 +0300
Subject: [PATCH 1/3] Avoid reparenting of InputPanel when the window is being
 destroyed

Fixes: QTBUG-95996
Pick-to: 5.15 6.1 6.2
Change-Id: Iac4d06cacf2e672bf4a7676f478798f27ab6f4a2
Reviewed-by: Jarkko Koivikko <jarkko.koivikko@code-q.fi>
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 0464e1bacca28d7e459f105b07fbea6bb5b20930)
---
 src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp b/src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp
index a7c0aad9..e04c828d 100644
--- a/src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp
+++ b/src/virtualkeyboard/qvirtualkeyboardinputcontext_p.cpp
@@ -278,8 +278,10 @@ void QVirtualKeyboardInputContextPrivate::onInputItemChanged()
                     For integrated keyboards, make sure it's a sibling to the overlay. The
                     high z-order will make sure it gets events also during a modal session.
                 */
-                if (isDesktopPanel.isValid() && !isDesktopPanel.toBool())
-                    vkbPanel->setParentItem(quickItem->window()->contentItem());
+                if (isDesktopPanel.isValid() && !isDesktopPanel.toBool()) {
+                    if (QQuickWindow *quickWindow = quickItem->window())
+                        vkbPanel->setParentItem(quickWindow->contentItem());
+                }
             }
         }
     } else {
-- 
2.35.1

